Sources/Commands/SwiftTestTool.swift

```diff
@@ -280,20 +288,51 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
                 numJobs: options.numberOfWorkers ?? ProcessInfo.processInfo.activeProcessorCount,
                 diagnostics: diagnostics,
                 options: self.options,
-                buildParameters: try self.buildParameters()
+                buildParameters: buildParameters
             )
             try runner.run(tests)
 
             if !runner.ranSuccessfully {
                 executionStatus = .failure
             }
         }
+
+        if options.shouldEnableCodeCoverage {
+            let buildParameters = try self.buildParameters()
+            let toolChain = try getToolchain()
+            try mergeRawDataFiles(fs: fileManager, path: buildParameters.buildPath.appending(component: "codecov"), toolChain: toolChain)
+            let graph = try loadPackageGraph()
+            for packages in graph.rootPackages {
+                let testPath = try buildTestsIfNeeded(options, graph: graph, binary: true)
```

Comment:
This doesn't make sense to me. We shouldn't call the buildTestsIfNeeded method here. This will end up running llbuild just to get the binary path. Also, getting the binary path like this is a bad idea. I would suggest refactoring this method to take the build plan so you can directory ask the binary path from it.

Reply:
done